DOCSP-41241: Laravel Sanctum (#3050)
jenssegers/laravel-mongodb: docs/user-authentication.txt

   :language: php
   :dedent:

Customize User Authentication
-----------------------------

You can customize your authentication files to align with your application's
needs and enable additional authentication features.

This section describes how to use the following features to customize the MongoDB user
authentication process:

- :ref:`laravel-user-auth-sanctum`
- :ref:`laravel-user-auth-reminders`

.. _laravel-user-auth-sanctum:

Laravel Sanctum
~~~~~~~~~~~~~~~

Laravel Sanctum is an authentication package that can manage API requests and
single-page application authentication. To manage API requests, Sanctum issues

> API tokens that are stored in a database table and authenticates incoming HTTP

Reviewer: Are you sure to keep the word "table", it refers to the SQL database.
```suggestion
API tokens that are stored in the database and authenticates incoming HTTP
```

   :language: php
   :dedent:

Customize User Authentication
-----------------------------

> In this section, you can learn how to use the following features to

Reviewer 2: S: before this sentence, add a sentence explaining why someone might want to make a custom model

customize your MongoDB user authentication process:

- :ref:`laravel-user-auth-sanctum`
- :ref:`laravel-user-auth-reminders`

.. _laravel-user-auth-sanctum:

Laravel Sanctum
~~~~~~~~~~~~~~~

> You can install the Laravel Sanctum package to provide API tokens to users

Reviewer 2: S: add a sentence explaining broadly what laravel sanctum is

and authenticate single-page applications. To manage API tokens, Sanctum stores

> each token in a database table and authenticates incoming HTTP requests through

Reviewer 2: ```suggestion
each token in a database table and authenticates incoming HTTP requests by using
```

the ``Authorization`` header. To authenticate single-page applications, Sanctum
uses Laravel's cookie-based authentication services.

Run the following commands from your project root to install Laravel Sanctum,
then publish and run its migration files:

.. code-block:: bash

   composer require laravel/sanctum
   php artisan vendor:publish --provider="Laravel\Sanctum\SanctumServiceProvider"

>    php artisan migrate

Reviewer: This last command cannot be done before the model is updated to use the `DocumentModel` trait.

.. tip::

>    To learn more about installing and configuring Laravel Sanctum, see the
>    `Installation <https://laravel.com/docs/{+laravel-docs-version+}/sanctum#installation>`__
>    section of the Laravel Sanctum guide.

Reviewer 2: ```suggestion
   To learn more about installing and configuring Laravel Sanctum, see
   `Installation <https://laravel.com/docs/{+laravel-docs-version+}/sanctum#installation>`__
   in the Laravel Sanctum guide.
```

To use Laravel Sanctum with {+odm-short+}, you must modify the ``PersonalAccessToken``

> model provided by Sanctum.

Reviewer 2: S: provide a little more explanation of what is required here

> Example

Reviewer: This is not an example but the required procedure.
